models/discrete/populations.py

```diff
@@ -347,55 +454,77 @@ def _compute_parameter_update_order(self):
                 if dependency in parameter_symbols:
                     G.add_edge(dependency, parsym)
                 elif dependency not in variable_symbols:
-                    raise PopulationSystemError(f"Parameter {parsym} references undefined symbol {dependency}")
+                    raise PopulationSystemError(
+                        f"Parameter {parsym} references undefined symbol {dependency}"
+                    )
         try:
             nodes = nx.topological_sort(G)
             ordered_parameters = [parameter_symbols[n] for n in nodes]
         except nx.exception.NetworkXUnfeasible:
-            raise PopulationSystemError(f"System parameters contain cyclic dependencies")
+            raise PopulationSystemError(
+                f"System parameters contain cyclic dependencies"
+            )
         return ordered_parameters
 
     def _compute_parameters(self):
```

Comment:
Could we have a test that checks that this method works as intended?

Reply:
Maybe the best way is to test against all possible 'correct' orderings for a test set?